feature/7573-UpdateAppColorsToComponentLibrary

theodur opened the pull request:
## Description of Change
<!-- Please include a description of the change and context. What would a code reviewer, or a future dev, 
need to know about this PR in order to understand why this PR was created? This could include dependencies 
introduced, changes in behavior, pointers to more detailed documentation. The description should be more 
than a link to an issue.  -->
Migrates the colors in the app to use the colors from the component library. Also removes unused color schemes from the repo.

## Screenshots/Video
<!-- Add screenshots or video as needed. Before/after if changes are to be compared by reviewers.
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Toggle: <details><summary></summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->
To many changes to show 😅

## Testing
<!-- What testing was done to verify the changes (local/unit)? What testing remains? Note edge cases, or special
situations that could not be tested during development. -->

- [x] Tested on iOS <!-- simulator is fine -->
- [x] Tested on Android <!-- simulator is fine -->

## Reviewer Validations
<!-- What should reviewers look for? Copy/paste Acceptance Criteria from ticket -->
Verify colors in the app have been migrated to use the component library colors [wherever applicable](https://docs.google.com/spreadsheets/d/146kWOH2rFeO63Y4Dy0MFL8yEn_4M7Cosc-zgbyWB3x8/edit?usp=sharing).

## PR Checklist
<!-- Engineer: make sure all these items are checked off before requesting a review -->
  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

TKDickson commented:
Colors are tough - I remember the original dark mode implementation and it was a LOT of work. Anyways, sending back for a greyer grey to @theodur

TimRoe commented on VAMobile/src/components/ActivityButton.tsx:
```diff
@@ -28,7 +26,7 @@ const ActivityButton: FC<ActivityButtonProps> = ({ title, subText, deepLink }: A
     backgroundColor: theme.colors.buttonBackground.activityButton as BackgroundVariant,
     style: {
-      shadowColor: colors.black,
+      shadowColor: 'black',
```
While looking at this PR to answer Ameet's question, just noting there is a black token if you wanted to lean on it instead of having some color strings.

theodur replied:
Good callout, updated this to use `vadsColorBlack`

TimRoe commented on VAMobile/src/constants/screens.ts:
```diff
@@ -23,7 +22,11 @@ export const LARGE_PANEL_OPTIONS: StackNavigationOptions = {
+        cardStyle: {
+          borderRadius: 0,
+          paddingTop: '30%',
+          backgroundColor: '#0000004D', // Black with 30% opacity
```
If it works the same, maybe preferable to use black + the [opacity](https://reactnative.dev/docs/view-style-props#opacity) prop.

theodur replied:
It's been a while, so I can't remember exactly, but I think the reason I had to set the opacity via the color's hex code is because when you set it manually via the `opacity` prop, weird things happen when a user has the screen reader enabled. I'll have to revisit this and see if that's still the case 🤔

brea11y reviewed (changes requested):
Hey @theodur – I noticed a couple of issues during the VQA and am listing them below (with screenshots). 

### Dark mode issues:
![IMG_0044](https://github.com/user-attachments/assets/e3c10c20-4d24-47bb-b5ce-af07621438d8)
The checkmark icon in picker list should be using vads-color-base-lightest. We've already adjusted that one in the Flagship library, so you can [preview how that should look here](https://www.figma.com/design/QVLPB3eOunmKrgQOuOt0SU/Flagship-Library---%F0%9F%93%90-Resource---VA-Mobile?node-id=9055-24306&t=VRvEihVdNLIgAHhs-4). It's basically just a lighter shade of gray than what I am seeing in the app.

![IMG_0042](https://github.com/user-attachments/assets/4808b0e3-92f9-47b4-8e8a-6c00c4dd8f67)
The radio buttons should have a dark border (vads-color-base-darkest), not a light border in dark mode. You can see how it should appear [in the FIgma library](https://www.figma.com/design/QVLPB3eOunmKrgQOuOt0SU/Flagship-Library---%F0%9F%93%90-Resource---VA-Mobile?node-id=750-7767&t=VRvEihVdNLIgAHhs-4).

![IMG_0043](https://github.com/user-attachments/assets/6207f673-67b8-4300-8791-5226b7e347b1)
In dark mode, the actual icons within the checkbox (the check mark in the example above or the hyphen for the partially selected) should be dark (vads-color-base-darkest), not light.

![IMG_0045](https://github.com/user-attachments/assets/98be88a3-a75f-488f-b252-5c6d6152a985)
I went through all of the different picker lists in the app and this one looked different in dark mode. All of the others ones looked as expected (see the first screenshot I added above) outside of some minor tweaks, but the copy in the selectable options all appears to be pulling over from light mode and not swapping to the lighter shade of gray that should be used (and is used throughout the rest of the app).

![IMG_0054](https://github.com/user-attachments/assets/1fc2599f-dde8-4019-b223-5d2805dbe5f7)
I believe the icons in the action sheet should be the same blue as what is used in the link copy.

![IMG_0049](https://github.com/user-attachments/assets/06201a44-432a-4eac-893f-446717e9f7da)
![IMG_0048](https://github.com/user-attachments/assets/37d7fbb8-27db-44de-a206-86a9f29157f6)
![IMG_0047](https://github.com/user-attachments/assets/b93438a7-08a8-4cc6-85e5-253ffcc238c5)
I believe that the check mark icons within the green circles are supposed to be dark gray (vads-color-base-darker). @jessicawoodin – Could you confirm?

![IMG_0046](https://github.com/user-attachments/assets/7434d34e-be01-4614-b27e-751832384325)
@jessicawoodin – The background color in the selected row also looks too light to me. Shouldn't this one be a darker shade of gray?

### Light mode issues:
![IMG_0059](https://github.com/user-attachments/assets/24d62b7f-8c8d-402d-bee7-44cd9035fc37)
![IMG_0061](https://github.com/user-attachments/assets/38176740-3c1f-429c-998a-4a886c91fdb3)
![IMG_0060](https://github.com/user-attachments/assets/d1e60175-cdad-415b-9d82-155eaa2d8184)
I believe the shade of green used for the circles around the check mark icons are supposed to be a lighter shade of green. I believe it should be uswds-system-color-green-cool-vivid-50. @jessicawoodin – Is that right?

![IMG_0055](https://github.com/user-attachments/assets/48e6194b-6803-4c50-b181-60db7837870a)
In the native alert, the link colors in light mode appear to be pulling the dark mode version of the links. They should be a darker blue (vads-color-primary-dark).

![IMG_0053](https://github.com/user-attachments/assets/73d3946b-b9ec-4214-9993-1e197ba3c3a8)
I believe the icons in the action sheet should match the color of the link (vads-color-primary-dark in light mode and uswds-system-color-blue-vivid-30 in dark mode).

![IMG_0052](https://github.com/user-attachments/assets/fca4507d-576b-415f-bcaa-6defe8196237)
In light mode, the solid circle icon that indicates a new message should be vads-color-primary-dark. Right now, it is using the shade of blue that should be used in dark mode.

![IMG_0057](https://github.com/user-attachments/assets/ec6894d9-be19-4dbf-8d90-f4a8e173f1b4)
In light mode, the action sheet appears to be pulling in the link colors from dark mode. The red should be vads-color-secondary-dark. The blue links should be vads-color-primary-dark.

jessicawoodin commented:
@brea11y Overall, this looks great! I agree with most of your feedback. To answer your questions:

- I believe that the check mark icons within the green circles are supposed to be dark gray (vads-color-base-darker).
     - You're correct — the checkmark inside the circle should appear dark gray. However, I think the issue is related to the actual icon being used. @theodur, can you confirm if the icons Brea flagged are coming from the design system or not? I think they may still be the older duotone icons from FontAwesome. If we update the icons to the design system icons, that will resolve this issue.
- The background color in the selected row also looks too light to me. Shouldn't this one be a darker shade of gray?
     - I compared this background color to what's currently in the app, and I think this color is technically correct as is.
- I believe the shade of green used for the circles around the check mark icons are supposed to be a lighter shade of green. I believe it should be uswds-system-color-green-cool-vivid-50.
     - Yup, that's correct! Those icons should be using uswds-system-color-green-cool-vivid-50 in light mode.

theodur commented:
@brea11y I updated/addressed the changes that weren't directly related to the icons, since we're still figuring out how to approach those.

> I went through all of the different picker lists in the app and this one looked different in dark mode. All of the others ones looked as expected (see the first screenshot I added above) outside of some minor tweaks, but the copy in the selectable options all appears to be pulling over from light mode and not swapping to the lighter shade of gray that should be used (and is used throughout the rest of the app).

I checked the color on my emulator, and it's showing the right color for that screen. Not sure why you were seeing the color used in light mode 🤔 here's a screenshot of what I'm seeing

<details> <summary> Screenshot </summary> 

![Screenshot_1730765549](https://github.com/user-attachments/assets/95d4457d-c30b-4814-a9f0-680bdbf5e71b)

</details>

I'm wondering if it was some type of glitch, since the colors for all selectors should be the same in the app. Could you check again and see if you're still seeing the color difference?

> In the native alert, the link colors in light mode appear to be pulling the dark mode version of the links. They should be a darker blue (vads-color-primary-dark).

We actually don't control the color of the native alert text in the app, so this one shouldn't be different from what's in prod.

> I believe the shade of green used for the circles around the check mark icons are supposed to be a lighter shade of green. I believe it should be uswds-system-color-green-cool-vivid-50. @jessicawoodin – Is that right?

I updated this to use `uswdsSystemColorGreenCoolVivid50` for light mode ✅

> In light mode, the action sheet appears to be pulling in the link colors from dark mode. The red should be vads-color-secondary-dark. The blue links should be vads-color-primary-dark.

I updated the blue link colors to be `vadsColorPrimaryDark`. I confirmed that the red links are actually already using `vadsColorSecondaryDark`

brea11y commented:
Hey @theodur – I just went through all of the changes that you mentioned above and they all look great, except for that one picker list. For whatever reason, it is still showing the light mode colors for the text in dark mode. I'm attaching another screenshot from iOS (it's not appearing that way on Android), but do completely agree that it seems odd and may, in fact, be a glitch. I even tried deleting the app completely and reinstalling from the new build that I created today and it still looks like the attached screenshot.
 <img width="369" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeEtmQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1044fdc7d6d8df467db1ef07f91cdbd82749fb45/IMG_0062.PNG" alt="IMG_0062.PNG" />

Another odd discrepency between iOS and Android that I'm now seeing is on the Android side. In light mode, the circle icon next to a new message is showing the darker shade of blue on both iOS and Android. In iOS, I am seeing the lighter shade of blue in dark mode (which is correct).. but now, in dark mode on Android, I'm seeing the darker shade of blue. Attaching a screenshot of that also. That seems super odd to me, so I wanted to make sure I sent it your way. I would think the colors would all pull in the same no matter which operating system you're using?
<img width="360" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeUNmQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--987d61bf9aedf859d16283a7d2400d4d7095a2c7/Screenshot_20241106_154033_VA.jpg" alt="Screenshot_20241106_154033_VA.jpg" />

theodur commented:
> Hey @theodur – I just went through all of the changes that you mentioned above and they all look great, except for that one picker list. For whatever reason, it is still showing the light mode colors for the text in dark mode. I'm attaching another screenshot from iOS (it's not appearing that way on Android), but do completely agree that it seems odd and may, in fact, be a glitch. I even tried deleting the app completely and reinstalling from the new build that I created today and it still looks like the attached screenshot.

@brea11y Figured out what was wrong and fixed the issue! Should be good now, let me know how that looks on your end

> Another odd discrepency between iOS and Android that I'm now seeing is on the Android side. In light mode, the circle icon next to a new message is showing the darker shade of blue on both iOS and Android. In iOS, I am seeing the lighter shade of blue in dark mode (which is correct).. but now, in dark mode on Android, I'm seeing the darker shade of blue. Attaching a screenshot of that also. That seems super odd to me, so I wanted to make sure I sent it your way. I would think the colors would all pull in the same no matter which operating system you're using?

I took a look at the colors used for the unread message circle icon, and confirmed they are what they should be, [vadsColorPrimary](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/feature/7573-UpdateAppColorsToComponentLibrary/VAMobile/src/styles/themes/colorSchemes.ts#L84) for light mode, and [uswdsSystemColorBlueVivid30](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/feature/7573-UpdateAppColorsToComponentLibrary/VAMobile/src/styles/themes/colorSchemes.ts#L226) for dark mode, which also aligns with the color of the chevron icon to the right of the message date. I also tried this on an Android device, and I'm seeing the correct lighter color in dark mode:

<details> 
<summary> Screenshot </summary>

![Screenshot_1731352876](https://github.com/user-attachments/assets/a65a48bb-b098-44f1-99a8-1d713b40350d)
</details>

I'm really not sure why the colors are showing different on your end 🤔. Is it this way on develop/the latest QA build?

brea11y commented:
Hey @theodur –

I'm not really sure why the new message indicator icon was being so glitchy, but everything looks great now in both dark and light mode. For some reason, that is the only area of the app that will not change without completely closing out the app. To force it to change to the right color, I have to close out of the app, switch the color mode from dark to light (or vice versa) and then launch the app again to see the change. For everything else, I can actually just change the color mode setting while I have the app open. However, since Veterans are likely never going to do that, I think this is fine. In the future, if we do decide to add the accessibility controls in the app for dark / light mode, we may want to circle back and see why this particular element isn't adjusting without completely closing out the app, but I think we're good for now.

I also double-checked the move menu / picker list in dark mode and in light mode (on Android and iOS even though Android was already showing properly) and it looks right on my end!

Since we are not going to be making adjustments to the icons in this ticket and will be revisiting those in another ticket, I'd say we should be good to wrap up VQA.

theodur commented:
@brea11y Thanks for confirming that we're all set! I took a look at production, and confirmed the same issue is happening there, with the unread message circle not adapting to the device's theme changing. This is due to the circle icon being rendered with the inbox list. So in order for the icon color to update, the inbox list would need to be updated, which explains why reopening the app displays the correct color. I agree that it's not likely that a user would be doing that mid app session, and even so, probably wouldn't notice. I can still make a bug ticket to look into it though.

Also, I when I did one final run through of the changes, I noticed the primary text color was off from one of my past commits. It was displaying the primary text in blue:

<details> <summary> Screenshot </summary>

![Simulator Screenshot - iPhone 13 - 2024-11-12 at 12 35 04](https://github.com/user-attachments/assets/7bf17a7d-ce3d-41f7-9f25-9ade1b9cd6ad)
</details>

I pushed a commit to fix that, so now the primary colors are normal again. Not sure whether you wanted to do another pass at VQA, but just wanted to let you know. I'll move this to `With QA` since everything looks good.

brea11y commented:
@theodur – Thanks so much for fixing this! I noticed the text color change in light mode, but assumed it was something else that had been requested as an update with another ticket. Things are constantly changing and I just figured someone else had requested that update and it was worked into these changes. 🤣 

I'll go ahead and run another build and do another look through all of the screens (including the onboarding carousel), just to make sure that there is nothing else that we missed and let you know ASAP. I don't anticipate any additional updates, but agree that it is better to take a last looks before we pass it on to QA.

theodur commented:
> Thanks so much for fixing this! I noticed the text color change in light mode, but assumed it was something else that had been requested as an update with another ticket. Things are constantly changing and I just figured someone else had requested that update and it was worked into these changes. 🤣

@brea11y That's what I thought at first too 😂 but when I did the comparison, I was like, "wait a minute..." lol

> I'll go ahead and run another build and do another look through all of the screens (including the onboarding carousel), just to make sure that there is nothing else that we missed and let you know ASAP. I don't anticipate any additional updates, but agree that it is better to take a last looks before we pass it on to QA.

I agree, that sounds good. Let me know if you notice anything!

theodur commented:
> Unfortunately, now the bullets in the onboarding carousel are not the right color in light mode. They are right in dark mode, but since we use the same colors in both dark and light mode for that carousel, let's make sure that both color modes are pulling in the bullet color that is currently being used for dark mode.

@brea11y Interestingly, this is how it is in prod right now too, which isn't correct. I think this change was missed in one of the recent icon tickets, most likely #9780. Even though it's not caused by the changes in this PR, I went ahead and made some changes to fix it; should be correct now ✅

In regard to the icons with contrast issues, I'm not sure why your icons are displaying like that, I just ran this branch on both iOS and Android and the icons appear the same as they do in production 🤔. This is what I'm seeing when I run the changes from this branch.

<details> <summary> Android Crisis Line screenshot </summary>

![Screenshot_1731456210](https://github.com/user-attachments/assets/7fe8d796-b915-4fe8-9c00-79de7116dfa3)
</details>

<details> <summary> iOS Crisis Line screenshot </summary>

![Simulator Screenshot - iPhone 13 - 2024-11-12 at 16 03 51](https://github.com/user-attachments/assets/b39289a4-1654-4bf3-8f51-54e2cbf279b8)
</details>

<details> <summary> Contact VA screenshot </summary>

![Simulator Screenshot - iPhone 13 - 2024-11-12 at 16 03 57](https://github.com/user-attachments/assets/a9a4f1eb-a7eb-44cf-96fc-c7df8599ba68)
</details>

Are you only noticing this issue on your Android device? I definitely agree that the contrast like that wouldn't be ideal, and also wouldn't be fulfilling the AC's of this ticket (to maintain parity with colors in production). Just not sure why they're displaying like that for you.

TKDickson commented:
Hey @brea11y @theodur - there's already a ticket for the onboarding carousel, and I think there's a fix in testable state anyways. No need to include in this PR. [10135](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/10135)

theodur commented:
> Hey @brea11y @theodur - there's already a ticket for the onboarding carousel bullet contrast, and I think there's a fix in testable state anyways. No need to include in this PR. [10135](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/10135)

@TKDickson Thanks for calling that out, didn't even realize. I already pushed a fix to my branch yesterday, I guess I'll revert that commit

brea11y commented:
@theodur – I had already run the build before you said that you'd revert the bullet back because of the other ticket. So, I do want to say that it looks great now (even though I know you'll be reverting it back).

Ugh.. Sorry, my brain is completely fried this week. I meant to include in my feedback yesterday that the icon issue is only occurring on Android. On iOS, I see what you are showing in your screenshots. I went ahead and created that new build today to check the bullet and take another look at the icons in iOS and Android. On iOS, I see the icons without the circles (except for the checkmark icons and those are the least problematic and not a huge issue) like you do, which is exactly how they should look. but even on that new build that I just created, the icons have the circles on Android and look like the screenshots I posted yesterday. In light mode, they have the darker shade of blue and white icons (which would be fine if it had to be that way) and in dark mode they have the light / pale blue circle and white icons. I even tried uninstalling the app completely and re-downloaded it from App Tester. I'm not sure what is going on and why they would be completely different?

brea11y commented:
This is approved on my end!

**Note for QA:** We did find a bug with the icons on Android. After doing some troubleshooting this afternoon with Theo, it appears that some Android users could see the old icons instead of the new icons (without the circle). My test device falls into that group, which is why things were looking so wonky on my end. Our hope is that the other ticket to update the icons and remove the duotone colors will resolve this issue when we get to it.

Huge thanks again @theodur for all of your hard work on this. I know it hasn't been easy and I really appreciate everything that you've done!

oddballdave commented:
Does this PR resolve this issue? https://github.com/department-of-veterans-affairs/va-mobile-app/issues/8798

TimRoe commented:
> Does this PR resolve this issue? #8798

No, this PR looks to have largely updated to _primitive_ tokens and not _semantic_ tokens. There's guidance on the difference [here on the doc site](https://department-of-veterans-affairs.github.io/va-mobile-app/design/Foundation/Design%20tokens/Color#how-to-use-color-tokens). Also the [design system Storybook colors page](https://department-of-veterans-affairs.github.io/va-mobile-library/?path=/docs/design-tokens-colors--docs) shows the primitive then semantic tokens.

oddballdave commented:
> #8798

Thanks for the clarification.
